pkg/apiserver/rest/dualwriter.go

```diff
@@ -65,8 +65,21 @@ type DualWriter struct {
 	legacy LegacyStorage
 }
 
-// NewDualWriter returns a new DualWriter.
-func NewDualWriter(legacy LegacyStorage, storage Storage) *DualWriter {
+type DualWriterMode int
+
+const (
+	Mode1 DualWriterMode = iota
+	Mode2
+	Mode3
+	Mode4
+)
+
+var CurrentMode = Mode2
```

Comment:
I would default the currentMode to 2 only when we have all the methods merged, wdyt?

Reply:
The current mode 2 behaviour falls back to the generic dual writer behaviour for everything except Create because most methods aren't defined in mode 2. Therefore, it's closer to the generic dual writer right now than when we will merge the remaining methods for mode 2.

One main difference in the Create method for mode 2 vs generic dual writing is that we now expect legacy storage to implement the `rest.Creater` instead of defaulting to just writing to Storage. I think that's a fair requirement.

The other one is that we are now setting additional fields in Storage (annotations and labels) but that shouldn't interfere with LegacyStorage Create.

It's also a safe mode to default to because it will continue reading and writing to LegacyStorage but additionally write to Storage on a best effort basis. The idea is to write to Storage when possible but not impede the LegacyStorage behavior. 

To summarise, dual writer mode 2 seems like the safest dual writer to default to besides the generic one. 

To be on the safer side, what I could do now is switch back to registering the generic dual writer and once we're further along and have more testing in place, we can switch it over.

Reply:
good point, so considering this I guess it's fine to merge things as they are, wdyt?